src/ui/dialogs/avatarupload.ts

```diff
@@ -0,0 +1,177 @@
+import Dialog from '../Dialog'
+import Client from '../../Client'
+
+let avatarUploadTemplate = require('../../../template/avatarUploadTemplate.hbs');
+
+let dialog: Dialog;
+let base64data : string;
+let mimetype : string;
+
+let height: string;
+let width: string;
+
+let account;
+
+export default function() {
+
+   let content = avatarUploadTemplate();
+
+   dialog = new Dialog(content);
+   let dom = dialog.open();
+
+   let accounts = Client.getAccountManager().getAccounts();
+   for (let i=0;i<accounts.length;i++){
+       let accountoption = $('<option value="'+i+'">'+accounts[i].getUid()+'</option>');
+       dialog.getDom().find('select[name="account"]').append(accountoption);
+   }
+
+   dialog.getDom().find('select[name="account"]').change((e)=>{
+       let val = Number($(e.target).val());
+       account=accounts[val];
+       loadAvatar(account,dialog);
+   });
+
+   console.log(mimetype);
+
+   $('.jsxc-avataruploadinput').change(function(event){
+        let file = (<HTMLInputElement>$('.jsxc-avataruploadinput')[0]).files[0];
+        handleFileSelect(file);
+   });
+
+  account=accounts[0];
+  loadAvatar(account,dialog);
+
+  dom.find('form').submit(onSubmit);
+
+  $('.jsxc-js-clear').on('click',removeAvatar);
+}
+
+function loadAvatar(account,dialog)
+{
+    account.getContact().getVcard().then(function(vcardData) {
+           let data = (vcardData as any);
+           if (data.hasOwnProperty('PHOTO'))
+           {
+              $('.jsxc-avatarimage > img').attr('src',data.PHOTO.src);
+              $('.jsxc-avatarimage > img').css('display','');
+           }
+    });
+}
+
+function handleFileSelect(file) {
+
+  if (!window.File || !window.FileReader || !window.FileList || !window.Blob) {
+    alert('The File APIs are not fully supported in this browser.');
+    return;
+  }
+
+  let fr = new FileReader();
+  fr.onload = receivedData;
+  //fr.readAsBinaryString(file)
+  fr.readAsDataURL(file);
+}
+
+function receivedData(result) {
+
+  let img =  <HTMLImageElement>$('.jsxc-avatarimage > img')[0];
+  $(img).css('height', '100%');
+  $(img).css('width', '100%');
+
+  img.onload = function() {
+
+    //https://xmpp.org/extensions/xep-0153.html#bizrules-image
+    //scale down to 192px
+    let result = scaleDownTo9kb(img, 192);
+
+    height = result.height.toString();
+    width  = result.width.toString();
+
+    base64data=result.base64;
+    mimetype=result.type;
+
+    $(img).attr('src', 'data:'+mimetype+';base64,'+base64data);
+    img.onload=null;
+    img.width=result.width;
+    img.height=result.height;
+    $(img).css('min-height', height);
+    $(img).css('min-width', width);
+    $(img).css('max-height', height);
+    $(img).css('max-width', width);
+    $(img).css('height', height);
+    $(img).css('width', width);
+    $(img).css('display','');
+  }
+  img.src = result.target.result;
+}
+
+function scaleDownTo9kb(img, start) : any
```

Comment:
Maybe we can move this to a utility class and use the same logic as the attachment.

Reply:
avatarupload.ts refactored...